app/invocation/invocation_action_card.tsx

```diff
@@ -127,65 +128,155 @@ export default class ActionCardComponent extends React.Component<Props, State> {
+                            Input Fetching Completed @{" "}
+                            {format.formatTimestampSecondsNanos(
+                              this.state.actionResult.executionMetadata.inputFetchCompletedTimestamp.seconds,
+                              this.state.actionResult.executionMetadata.inputFetchCompletedTimestamp.nanos
+                            )}
+                          </div>
+                          <div className="metadata-detail">
+                            Execution Started @{" "}
+                            {format.formatTimestampSecondsNanos(
+                              this.state.actionResult.executionMetadata.executionStartTimestamp.seconds,
+                              this.state.actionResult.executionMetadata.executionStartTimestamp.nanos
+                            )}
+                          </div>
+                          <div className="metadata-detail">
+                            Execution Completed @{" "}
+                            {format.formatTimestampSecondsNanos(
+                              this.state.actionResult.executionMetadata.executionCompletedTimestamp.seconds,
+                              this.state.actionResult.executionMetadata.executionCompletedTimestamp.nanos
+                            )}
+                          </div>
+                          <div className="metadata-detail">
+                            Output Upload Started @{" "}
+                            {format.formatTimestampSecondsNanos(
+                              this.state.actionResult.executionMetadata.outputUploadStartTimestamp.seconds,
+                              this.state.actionResult.executionMetadata.outputUploadStartTimestamp.nanos
+                            )}
+                          </div>
+                          <div className="metadata-detail">
+                            Output Upload Completed @{" "}
+                            {format.formatTimestampSecondsNanos(
+                              this.state.actionResult.executionMetadata.outputUploadCompletedTimestamp.seconds,
+                              this.state.actionResult.executionMetadata.outputUploadCompletedTimestamp.nanos
+                            )}
+                          </div>
+                          <div className="metadata-info">
+                            Worker Completed @{" "}
+                            {format.formatTimestampSecondsNanos(
+                              this.state.actionResult.executionMetadata.workerCompletedTimestamp.seconds,
+                              this.state.actionResult.executionMetadata.workerCompletedTimestamp.nanos
+                            )}
+                          </div>
+                        </div>
+                      ) : (
+                        <div>None found.</div>
+                      )}
+                    </div>
+                    <div className="action-section">
```

Comment:
Commands also have an input directory digest -- we might not want to fetch the complete tree of inputs and show it in the UI in this PR, but maybe we could at least show the total size of inputs just above this? (You could get that as the size field of the input root digest field)